ports/spresense/Makefile

```diff
@@ -0,0 +1,186 @@
+	--defsym __stack=_vectors+786432 \
+	-T$(SPRESENSE_SDK)/nuttx/build/ramconfig.ld \
+	--gc-sections \
+	-Map=$(BUILD)/output.map \
+	-o $(BUILD)/circuitpython.elf \
+	--start-group \
+	-u spresense_main \
+	$(BUILD)/libmpy.a \
+	$(SPRESENSE_SDK)/sdk/libs/libapps.a \
+	$(SPRESENSE_SDK)/sdk/libs/libsdk.a \
+	$(LIBM) \
+	$(LIBGCC) \
+	--end-group \
+	-L$(BUILD) \
+
+CFLAGS += -DCFG_TUSB_MCU=OPT_MCU_CXD56 -DCFG_TUD_MIDI_RX_BUFSIZE=128 -DCFG_TUD_CDC_RX_BUFSIZE=256 -DCFG_TUD_MIDI_TX_BUFSIZE=128 -DCFG_TUD_CDC_TX_BUFSIZE=256 -DCFG_TUD_MSC_BUFSIZE=512 $(CFLAGS_MOD)
+
+SRC_COMMON_HAL_EXPANDED = $(addprefix shared-bindings/, $(SRC_COMMON_HAL)) \
+                          $(addprefix shared-bindings/, $(SRC_BINDINGS_ENUMS)) \
+                          $(addprefix common-hal/, $(SRC_COMMON_HAL))
+
+SRC_SHARED_MODULE_EXPANDED = $(addprefix shared-bindings/, $(SRC_SHARED_MODULE)) \
+                             $(addprefix shared-module/, $(SRC_SHARED_MODULE))
+
+SRC_S = supervisor/cpu.s
+
+SRC_C = \
+	tick.c \
+	background.c \
+	fatfs_port.c \
+	mphalport.c \
+	lib/utils/stdout_helpers.c \
+	lib/utils/pyexec.c \
+	lib/libc/string0.c \
+	lib/mp-readline/readline.c \
+	lib/timeutils/timeutils.c \
+	lib/oofatfs/ff.c \
+	lib/oofatfs/option/ccsbcs.c \
+	lib/utils/interrupt_char.c \
+	lib/utils/sys_stdio_mphal.c \
+	lib/utils/context_manager_helpers.c \
+	lib/utils/buffer_helper.c \
+	supervisor/shared/memory.c \
+	lib/tinyusb/src/portable/sony/cxd56/dcd_cxd56.c \
+
+OBJ = $(PY_O) $(SUPERVISOR_O) $(addprefix $(BUILD)/, $(SRC_C:.c=.o))
+OBJ += $(addprefix $(BUILD)/, $(SRC_S:.s=.o))
+OBJ += $(addprefix $(BUILD)/, $(SRC_COMMON_HAL_EXPANDED:.c=.o))
+OBJ += $(addprefix $(BUILD)/, $(SRC_SHARED_MODULE_EXPANDED:.c=.o))
+
+# List of sources for qstr extraction
+SRC_QSTR += $(SRC_C) $(SRC_SUPERVISOR) $(SRC_COMMON_HAL_EXPANDED) $(SRC_SHARED_MODULE_EXPANDED)
+# Sources that only hold QSTRs after pre-processing.
+SRC_QSTR_PREPROCESSOR += 
+
+all: $(BUILD)/circuitpython.spk
+
+$(FIRMWARE):
+	$(ECHO) ""
+	$(ECHO) "Download the spresense binaries zip archive from:"
+	$(ECHO) "https://developer.sony.com/file/download/download-spresense-firmware-v1-4-000"
+	$(ECHO) "Extract spresense binaries to $(FIRMWARE)"
+	$(ECHO) ""
+	$(ECHO) "run make flash-bootloader again to flash bootloader."
+	exit 1
+
+$(BUILD)/libmpy.a: $(SPRESENSE_SDK) $(OBJ) 
+	$(ECHO) "AR $@"
+	$(Q)$(AR) rcs $(BUILD)/libmpy.a $(OBJ)
+
+$(BUILD)/circuitpython.elf: $(BUILD)/libmpy.a
+	$(ECHO) "LD $@"
+	$(Q)$(LD) $(LDFLAGS)
+
+$(BUILD)/circuitpython.spk: $(BUILD)/circuitpython.elf
+	$(ECHO) "Creating $@"
+	$(SPRESENSE_SDK)/sdk/tools/linux/mkspk -c 2 $(BUILD)/circuitpython.elf nuttx $(BUILD)/circuitpython.spk
+
+flash: $(BUILD)/circuitpython.spk
+	$(ECHO) "Writing $< to the board"
+	$(SPRESENSE_SDK)/sdk/tools/flash.sh -c /dev/ttyUSB0 $(BUILD)/circuitpython.spk
+
+flash-bootloader: $(SPRESENSE_SDK) $(FIRMWARE)
+	$(ECHO) "Writing loader to the board"
+	$(SPRESENSE_SDK)/sdk/tools/flash.sh -l $(FIRMWARE) -c /dev/ttyUSB0
```

Comment:
I wasn't able to do this step on Mac. The kernel spun up for a while and then the script produced this error:

```
> spresense-exported-sdk/sdk/tools/flash.sh -l spresense-exported-sdk/firmware -c /dev/tty.usbmodemE594E79FECF32
Traceback (most recent call last):
  File "site-packages/pyserial-2.7-py3.6.egg/serial/serialposix.py", line 468, in read
OSError: [Errno 6] Device not configured

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "flash_writer.py", line 550, in <module>
  File "flash_writer.py", line 490, in main
  File "flash_writer.py", line 357, in cancel_autoboot
  File "flash_writer.py", line 286, in readline
  File "site-packages/pyserial-2.7-py3.6.egg/serial/serialposix.py", line 480, in read
TypeError: 'OSError' object is not subscriptable
[60064] Failed to execute script flash_writer
```

Any suggestions? Do I need to make sure jumpers are in a particular spot? ladyada may have used the board before me.

Reply:
1. Have you installed the serial driver?
2. Have you connected USB to main board? USB connected to the extension board is not intended for flashing new image.
3. Try using /dev/tty.SLAB_USBtoUART as a serial device.

Reply:
I hadn't updated the serial driver and may have been using the wrong serial device. I got it working today though! 🎉